src/update.cpp

```diff
@@ -245,7 +259,8 @@ Rcpp::newDatetimeVector C_update_dt(const Rcpp::NumericVector& dt,
   }
   load_tz_or_fail(tzto, tzone2, "CCTZ: Unrecognized tzone: \"%s\"");
 
-  Rcpp::NumericVector out(N);
+  cpp11::writable::doubles out(N);
```

Comment:
Why do you use `writable` on output vectors?  Can you even return cpp11::doubles back to R? Is there a performance penalty in writable constructor?

Reply:
> Can you even return cpp11::doubles back to R

I'm fairly certain that you can, but you can't modify them at the C++ level using the cpp11 API

> Is there a performance penalty in writable constructor?

No, AFAIK it just exposes the ability write to elements with syntax like `out[i] = value`. The constructor itself essentially just eventually calls this reserve method that calls `Rf_allocVector()` and then shoves the allocated memory in a doubly linked list object that can preserve it
https://github.com/r-lib/cpp11/blob/5892862b77b59e4a443e4c58320e5496530e9a30/inst/include/cpp11/doubles.hpp#L104-L114

A few things you may find interesting
https://cpp11.r-lib.org/articles/converting.html#class-comparison-table
https://cpp11.r-lib.org/articles/converting.html#read-only-vs-writable-vectors-1
https://cpp11.r-lib.org/articles/converting.html#fewer-implicit-conversions-1

The process I went through to convert lubridate over to cpp11
https://cpp11.r-lib.org/articles/converting.html#mechanics-of-converting-a-package-from-rcpp-1

Reply:
Ok, I got it.